src/nvim/api/ui.c

```diff
@@ -273,6 +248,39 @@ void nvim_ui_set_option(uint64_t channel_id, String name,
   }
   UI *ui = pmap_get(uint64_t)(connected_uis, channel_id);
 
+  if (strequal(name.data, "term_name")) {
+    if (value.type != kObjectTypeString) {
+      api_set_error(error, kErrorTypeValidation, "term_name must be a String");
+      return;
+    }
+    set_tty_option("term", xstrdup(value.data.string.data));
+    return;
+  }
+  if (strequal(name.data, "term_colors")) {
+    if (value.type != kObjectTypeInteger) {
+      api_set_error(error, kErrorTypeValidation, "term_colors must be a Integer");
+      return;
+    }
+    t_colors = (int)value.data.integer;
+    return;
+  }
+  if (strequal(name.data, "term_background")) {
+    if (value.type != kObjectTypeString) {
+      api_set_error(error, kErrorTypeValidation, "term_background must be a String");
+      return;
+    }
+    if (option_was_set("bg")
+        || strequal((char *)p_bg, value.data.string.data)) {
+      api_set_error(error, kErrorTypeValidation, "background is already set");
```

Comment:
This is not an error. We should just silently ignore it when the user already has set 'bg'. The terminal cannot know if the user has done it or not.